exchanges/btse/btse_test.go

```diff
@@ -977,52 +702,41 @@ func TestGetLatestFundingRates(t *testing.T) {
 		Pair:                 currency.NewPair(currency.BTC, currency.USDT),
 		IncludePredictedRate: true,
 	})
-	if !errors.Is(err, asset.ErrNotSupported) {
-		t.Error(err)
-	}
+	assert.ErrorIs(t, err, asset.ErrNotSupported, "GetLatestFundingRates should error on Margin")
+
 	_, err = b.GetLatestFundingRates(context.Background(), &fundingrate.LatestRateRequest{
 		Asset: asset.Futures,
 	})
-	if err != nil {
-		t.Error(err)
-	}
+	assert.NoError(t, err, "GetLatestFundingRates should not error on futures")
 
 	_, err = b.GetLatestFundingRates(context.Background(), &fundingrate.LatestRateRequest{
 		Asset: asset.Futures,
-		Pair:  testFUTURESPair,
+		Pair:  futuresPair,
 	})
-	if err != nil {
-		t.Error(err)
-	}
+	assert.NoError(t, err, "GetLatestFundingRates should not error on futures")
 }
 
 func TestIsPerpetualFutureCurrency(t *testing.T) {
 	t.Parallel()
 	isPerp, err := b.IsPerpetualFutureCurrency(asset.CoinMarginedFutures, currency.NewPair(currency.BTC, currency.USD))
-	if err != nil {
-		t.Error(err)
-	}
-	if isPerp {
-		t.Error("expected false")
-	}
+	assert.NoError(t, err, "IsPerpetualFutureCurrency should not error")
+	assert.False(t, isPerp, "IsPerpetualFutureCurrency should return true for a Margin pair")
 
-	isPerp, err = b.IsPerpetualFutureCurrency(asset.Futures, testFUTURESPair)
-	if err != nil {
-		t.Error(err)
-	}
-	if !isPerp {
-		t.Error("expected true")
-	}
+	isPerp, err = b.IsPerpetualFutureCurrency(asset.Futures, futuresPair)
+	assert.NoError(t, err, "IsPerpetualFutureCurrency should not error")
+	assert.True(t, isPerp, "IsPerpetualFutureCurrency should return true for a futures pair")
 
-	cp, err := currency.NewPairFromString(testSPOTPair)
-	if err != nil {
-		t.Error(err)
-	}
-	isPerp, err = b.IsPerpetualFutureCurrency(asset.Futures, cp)
-	if err != nil {
-		t.Error(err)
-	}
-	if isPerp {
-		t.Error("expected false")
-	}
+	isPerp, err = b.IsPerpetualFutureCurrency(asset.Futures, spotPair)
+	assert.NoError(t, err, "IsPerpetualFutureCurrency should not error")
+	assert.False(t, isPerp, "IsPerpetualFutureCurrency should return false for a spot pair")
+}
+
+var updatePairsGuard sync.Once
+
+func updatePairsOnce(tb testing.TB) {
+	tb.Helper()
+	updatePairsGuard.Do(func() {
+		err := b.UpdateTradablePairs(context.Background(), true)
+		assert.NoError(tb, err, "UpdateTradablePairs should not error")
+	})
```

Comment:
I've left this in because I've used it for an improved test around using pairs from GetEnabledPairs() for ticker and orderbook tests to ensure futures pairs are useable directly.
I've moved that work out for now, but feel this is useful enough to remain lurking for when I pick that back up.

Reply:
I like this a lot. The multiple calls to `UpdateTradablePairs` slows down testing